elastic/kibana
# [SIEM][Lists] Adds 90% of the REST API and client API for exception lists and exception items

## Summary

See for more details:
https://github.com/elastic/kibana/issues/65938

Adds pieces of the `exception list` and `exception list item` and refactors/cleans the code up where I had parts incorrect with little things such as the javascript library io-ts. Some unit tests were added but I am holding off until more of the operations solidify before adding the unit tests. Everything is still behind a feature flag that must be enabled and not advised still at this point to use so I feel ok pushing these parts forward.

Adds to the API:
- Create exception list
- Read exception list
- Update exception list
- Delete exception list (and exception list items that are associated with it)
- Create exception list item
- Find exception list (/_find)
- Read exception list item
- Update exception list item
- Delete exception list items individually
- Find exception list item (/_find)

What is still missing from the REST and client API?
- Patch exception list
- Patch exception list item
- Bulk versions of everything
- Import/Export options for these exception lists and list items

### Manual testing and REST API endpoints

Go here:
```sh
/projects/kibana/x-pack/plugins/lists/server/scripts
```

See the files:

```sh
delete_all_exception_lists.sh
delete_exception_list.sh
delete_exception_list_by_id.sh
delete_exception_list_item.sh
delete_exception_list_item_by_id.sh
exception_lists
find_exception_list_items.sh
find_exception_lists.sh
get_exception_list.sh
get_exception_list_by_id.sh
get_exception_list_item.sh
get_exception_list_item_by_id.sh
post_exception_list.sh
post_exception_list_item.sh
update_exception_list.sh
update_exception_list_item.sh
```

Ensure you first run:

```sh
./hard_reset
```

and ensure you have setup your kibana.dev.yml to have:

```yml
# Enable lists feature
xpack.lists.enabled: true
xpack.lists.listIndex: '.lists-frank'
xpack.lists.listItemIndex: '.items-frank'
```

Then you can use the above scripts to create, read, update, and delete exception list and exception list items as well as perform find commands against them all.

### Checklist

Delete any items that are not applicable to this PR.


Note: Some but limited unit tests at this point.
